### PR TITLE
DMW-Thesis

### PR DESCRIPTION
**Pull request by Odin-byte**

### CODE REVIEW

File: robmuxinator/robmuxinator.py [hunk 1]
```diff
@@ -763,6 +763,12 @@ def main():
         help="the path to the yaml config file",
         default="/etc/ros/upstart_robot.yaml",
     )
+    parser.add_argument(
+        "-n",
+        "--number",
+        help="the number to append to each session name within the config file. used for multi robot setups. Can also be used within sessions as a env_variable. this is used as a unique ros domain id",
+        default="",
+    )
```
Comment (fmessmer) [inline robmuxinator/robmuxinator.py:771]: I'm a bit hesitant to just call this `number`
Maybe we should rather call this `session_id_no` or `session_prefix_no` or something

We should maybe also parse this argument as an integer explicitly to have some type checking

It is later used as `ROS_DOMAIN_ID` - is this really guaranteed to be unique?

**Review by fmessmer — CHANGES_REQUESTED** [review]

File: robmuxinator/robmuxinator.py (hunk 1 above)
Comment (fmessmer) [reply robmuxinator/robmuxinator.py:771]: https://github.com/4am-robotics/orga/wiki/ROS-Domain-Ids starts with `7`
I'll add `reserved` for `0-6` so no one is going to assign those `ROS_DOMAIN_ID`s

:point_right: https://github.com/4am-robotics/orga/wiki/ROS-Domain-Ids/3025d5cc6f7d29fcc6cc63d32e79c96a6bffd74a

Comment (fmessmer) [reply robmuxinator/robmuxinator.py:771]: or maybe uuid?

Comment (Odin-byte) [reply robmuxinator/robmuxinator.py:771]: I would lean towards session_prefix_no

Comment (fmessmer) [reply robmuxinator/robmuxinator.py:771]: `instance_id` it is - as discussed with @Odin-byte

**Comment by fmessmer** [comment]
the (merge-able) version is provided via https://github.com/4am-robotics/robmuxinator/pull/19
in order to not conflict with @Odin-byte thesis work

**Comment by fmessmer** [comment]
I guess we can close this PR in favor of https://github.com/4am-robotics/robmuxinator/pull/19

**Pull request closed without merging**